keps/sig-auth/3299-kms-v2-improvements/README.md

```diff
@@ -396,16 +401,29 @@ FeatureSpec{
 }
 ```
 
+**Beta**
+```go
+FeatureSpec{
+	Default: true,
+	LockToDefault: false,
+	PreRelease: featuregate.Beta,
+}
+```
+
 ###### Does enabling the feature change any default behavior?
 
-No. The v2 API is new in the v1.25 release.
+No. The v2 API is new in the v1.25 release. Furthermore, even with the feature enabled by default, the user needs to explicitly configure a KMSv2 provider to use this.
 
 ###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
 
 Yes, To disable encryption at rest using the v2 API:
-1. Disable encryption at rest with KMS provider by running through these [steps](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/#disabling-encryption-at-rest)
-   1. At the end of this step, all the data in etcd will be unencrypted.
-2. Disable the `KMSv2` feature gate.
+1. Add new `identity` provider at the top of encryption config
+1. Restart kube-apiserver
+1. Run storage migration to migrate all the existing encrypted data to use the `identity` provider
+1. Remove the KMS provider from the encryption config and restart kube-apiserver
+2. At the end of these steps, all the data in etcd will be unencrypted.
+
+More details are available [here](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/#disabling-encryption-at-rest)
 
 Disabling this gate without first doing a storage migration to use a different encryption at rest mechanism will result in data loss.
```

Comment:
> It's not just data loss, right? It will basically break the cluster, since apiserver won't be able to read the objects from etcd.

That is only when all the resources in etcd are encrypted using KMS. Typically the configuration is only for a subset of resources (secrets, configmaps). In that scenario, those resources can't be retrieved if the KMS providers are removed without running storage migration. If the user adds the KMS providers back (only for read), they would still be able to retrieve the resources.

Reply:
Let's imagine we encrypted secrets that are mounted in pods. This will prevent any future pods referencing secrets from starting, right?  Given that secrets are the most likely thing to encrypt, it's worth calling out the consequences of the failure here.